scripts/game_structure/game/save_load/save_load.py

```diff
@@ -0,0 +1,123 @@
+import os
+from shutil import move as shutil_move
+
+import ujson
+
+from scripts.housekeeping.datadir import get_temp_dir, get_save_dir
+
+
+def safe_save(path: str, write_data, check_integrity=False, max_attempts: int = 15):
+    """If write_data is not a string, assumes you want this
+    in json format. If check_integrity is true, it will read back the file
+    to check that the correct data has been written to the file.
+    If not, it will simply write the data to the file with no other
+    checks."""
+
+    # If write_data is not a string,
+    if type(write_data) is not str:
+        _data = ujson.dumps(write_data, indent=4)
+    else:
+        _data = write_data
+
+    dir_name, file_name = os.path.split(path)
+
+    if check_integrity:
+        if not file_name:
+            raise RuntimeError(f"Safe_Save: No file name was found in {path}")
+
+        temp_file_path = get_temp_dir() + "/" + file_name + ".tmp"
```

Comment:
Moving forward, we should try to use Pathlib instead of constructing path literals as strings.